### PR TITLE
Improve function `pert_scaled`

### PR DESCRIPTION
**Pull request by thangckt**
My suggestion:
- Separate VASP and ABACUS into separated functions (and may be separated files). This will helpful to someone who want to implement a new DFT engine and also convenient for maintain.

## Summary by CodeRabbit

- **New Features**
  - Enhanced the `pert_scaled` function to support different initialization styles ("VASP" or "ABACUS").
  
- **Refactor**
  - Improved loop logic and file handling for better performance and reliability in perturbation processes.
  - Streamlined the `make_vasp_relax` and `make_scale` functions for improved readability and clarity.
  - Simplified the construction of perturbation commands in the `pert_scaled` function.

### CODE REVIEW

File: dpgen/data/gen.py [hunk 1]
```diff
@@ -805,35 +808,7 @@ def pert_scaled(jdata):
                 else:
                     shutil.copy2(pos_in, pos_out)
                 os.remove(pos_in)
-            kk = -1
```
Comment (Liu-RX) [inline dpgen/data/gen.py:808]: So long as these codes are deleted, where are the new codes corresponding to them?

Comment (thangckt) [reply dpgen/data/gen.py:808]: They are no longer needed. Please let the others know the new code work or not?

**Review by njzjz — COMMENTED** [review]
I am not familiar with this piece of code. I can merge if @Liu-RX approves the PR.

**Comment by njzjz** [comment]
@Liu-RX could you take a review? Thanks!

File: dpgen/data/surf.py [hunk 2]
```diff
@@ -23,7 +23,7 @@
 import dpgen.data.tools.fcc as fcc
 import dpgen.data.tools.hcp as hcp
 import dpgen.data.tools.sc as sc
-from dpgen import ROOT_PATH, dlog
```
Comment (Liu-RX) [inline dpgen/data/surf.py:26]: Is removing ROOT_PATH necessary? Is ROOT_PATH used elsewhere?

File: dpgen/data/surf.py [hunk 3]
```diff
@@ -440,15 +441,12 @@ def make_scale(jdata):
         for jj in scale:
             if skip_relax:
                 pos_src = os.path.join(os.path.join(init_path, ii), "POSCAR")
-                assert os.path.isfile(pos_src)
```
Comment (Liu-RX) [inline dpgen/data/surf.py:443]: Why remove this assert? I don't understand.

**Comment by Liu-RX** [comment]
> @Liu-RX could you take a review? Thanks!

Sorry for the late reply!!!

File: dpgen/data/surf.py (hunk 3 above)
Comment (thangckt) [reply dpgen/data/surf.py:445]: If you read some more lines, you will is that assert is redundant

File: dpgen/data/surf.py (hunk 2 above)
Comment (thangckt) [reply dpgen/data/surf.py:21]: It is not be used

**Review by Liu-RX — APPROVED** [review]

**Pull request merged**